internal/provider/resource_saml_group_mapping.go

```diff
@@ -0,0 +1,380 @@
+package provider
+
+import (
+	"context"
+	"errors"
+	"slices"
+	"strings"
+
+	"github.com/google/uuid"
+	"wiz.io/hashicorp/terraform-provider-wiz/internal/utils"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"wiz.io/hashicorp/terraform-provider-wiz/internal"
+	"wiz.io/hashicorp/terraform-provider-wiz/internal/client"
+	"wiz.io/hashicorp/terraform-provider-wiz/internal/wiz"
+)
+
+// ReadSAMLGroupMappings represents the structure of a SAML group mappings read operation.
+// It includes a SAMLGroupMappings object.
+type ReadSAMLGroupMappings struct {
+	SAMLGroupMappings SAMLGroupMappings `json:"samlIdentityProviderGroupMappings"`
+}
+
+// SAMLGroupMappings represents the structure of SAML group mappings.
+// It includes PageInfo and a list of Nodes.
+type SAMLGroupMappings struct {
+	PageInfo wiz.PageInfo            `json:"pageInfo"`
+	Nodes    []*wiz.SAMLGroupMapping `json:"nodes,omitempty"`
+}
+
+// SAMLGroupMappingsImport represents the structure of a SAML group mapping import.
+// It includes the SAML IdP ID, provider group ID, project IDs, and role.
+type SAMLGroupMappingsImport struct {
+	SamlIdpID       string
+	ProviderGroupID string
+	ProjectIDs      []string
+	Role            string
+}
+
+func resourceWizSAMLGroupMapping() *schema.Resource {
+	return &schema.Resource{
```

Comment:
The correct term is SAML assertion, update can be made to "_Configure SAML Group Role Mapping. When using SSO to authenticate with Wiz, you can map group memberships in SAML assertions to Wiz roles across specific scopes._"